packages/build/bin/compile-package.js

```diff
@@ -128,7 +134,69 @@ function run(argv, options) {
 
   args.push(...compilerOpts);
 
-  return utils.runCLI('typescript/lib/tsc', args, {cwd, ...options});
+  const validArgs = validArgsForBuild(args);
+
+  return utils.runCLI('typescript/lib/tsc', validArgs, {cwd, ...options});
+}
+
+/**
+ * `tsc -b` only accepts valid arguments. `npm run build -- --<other-arg>` may
+ * pass in extra arguments. We need to remove such arguments.
```

Comment:
What kind of arguments are getting in via `npm run build -- --<other-arg>`? If they are provided by the user (either on command line or via their package.json) then we must let them know about the problem and especially which parameters are removed by `lb-tsc`.

Personally, I don't think it's a good idea to add such parameter-modifications to `lb-tsc`, I am concerned about support and maintenance costs. IMO, we should let `tsc` fail the build and expect the user to fix the parameters passed to`lb-tsc`/`tsc`.